microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -0,0 +1,92 @@
+)
+class BVTExtension(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        creates a restore point collection then a restore point which helps in
+        validating the VMSnapshot extension.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_vmsnapshot_extension(self, log: Logger, node: Node,
+                                    environment: Environment) -> None :
+        unique_name = str(uuid.uuid4())
+        information = environment.get_information()
+        resource_group_name = information["resource_group_name"]
+        location = information["location"]
+        vm_name = node.name
+        log.info(f"information {information}")
+        restore_point_collection = "rpc_" + unique_name
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+        sub_id = platform.subscription_id
+        # creating restore point collection
+        client = get_compute_client(environment.platform)
+        response = client.restore_point_collections.create_or_update(
+            resource_group_name=information["resource_group_name"],
+            restore_point_collection_name=restore_point_collection,
+            parameters={
+                "location": location,
+                "properties": {
+                    "source": {
+                        "id": "/subscriptions/" + sub_id + "/resourceGroups/" + resource_group_name + "/providers/Microsoft.Compute/virtualMachines/" + vm_name
+                    }
+                },
+            },
+        )
+        log.info("rpc created")
+        log.info(f"response {response}")
+        count = 0
+
+        while count < 10:
+            vm = client.virtual_machines.get(
+                resource_group_name,
+                vm_name
+            )
+            # check the state of the VM
+            if vm.provisioning_state == "Succeeded":
+                try:
+                    # create a restore point for the VM
+                    response = client.restore_points.begin_create(
+                        resource_group_name=information["resource_group_name"],
+                        restore_point_collection_name=restore_point_collection,
+                        restore_point_name="rp_" + datetime.now().strftime("%Y-%m-%d-%H-%M-%S"),
+                        parameters={},
+                    )
+                    response.wait(3600)
+                    log.info("restore point created")
+                    break
+                except Exception as e:
+                    if "Changes were made to the Virtual Machine" in str(e):
+                        pass
+                    else:
+                        log.info(f"error {e}")
+                        raise AssertionError("Test failed: Unexpected error occurred")
+            time.sleep(1)
+            count = count + 1
+        assert count < 10, "Failed in Creating Restore Point"
```

Comment:
1. use assertpy, it shows original values. Refer to examples in code about how to assert an integer.
2. It looks not necessary. If count is smaller than 10, either in the loop, or raised exception.

Reply:
This was used to fail the test case when the count >= 10 which means it has retried 10 times but still was not successful in creating a restore point. (This can happen when it do not get into the if block if vm.provisioning_state == "Succeeded": i.e VM_state is not ready and keeps waiting for it.)
As you have mentioned the VM would always be in ready state will remove this check.

Reply:
Will continue to have the check as if say 10 times "Changes were made to the Virtual Machine" error occurred then it would just simply pass and when exceeded 10 retries it just comes out of the loop passing the test, so to avoid it and trying to fail the test